### PR TITLE
[DF-489] BE: Stellebewirtschaften - added search highlighting, status…

### PR DESCRIPTION
**Pull request by wwebec**
… filter and tests

### CODE REVIEW

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
     private String keywordsText;
 
+    private Boolean displayRestricted;
```
Comment (pavax) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java]: what I mean't is that on the ManagedJobAdSearchRequest we should have a property called `state`  where we could filter for job's in that given state.

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java [hunk 2]
```diff
@@ -17,6 +17,8 @@
 
     private String keywordsText;
 
+    private String state;
```
Comment (pavax) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java]: why is it a string?

Comment (wwebec) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java]: changed state type of ManagedJobAdSearchRequest to JobAdvertisementStatus

File: infrastructure-elasticsearch/src/main/resources/config/elasticsearch/mappings/job-advertisement.json [hunk 3]
```diff
@@ -290,6 +290,9 @@
           },
           "stellennummerEgov": {
             "type": "keyword"
+          },
+          "stellennummerAvam": {
+            "type": "text"
```
Comment (pavax) [inline infrastructure-elasticsearch/src/main/resources/config/elasticsearch/mappings/job-advertisement.json]: why is the stellennummerAvam text but the stellennummerEgov keyword?

Comment (wwebec) [reply infrastructure-elasticsearch/src/main/resources/config/elasticsearch/mappings/job-advertisement.json]: Initially I had set an index stellennummerAvam" as "keyword" however (at least locally) it was recognized as an attempt of updating of existing mappings which caused the following error:
```java
 Error creating bean with name 'elasticsearchIndexService' defined in file [...\ElasticsearchIndexService.class]: Unsatisfied dependency expressed through constructor parameter 2; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jobAdvertisementElasticsearchRepository': Invocation of init method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository]: Constructor threw exception; nested exception is java.lang.IllegalArgumentException: mapper [jobAdvertisement.stellennummerAvam] of different type, current_type [text], merged_type [keyword]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:732)`

**Pull request merged**